servers/zms/src/main/java/com/yahoo/athenz/zms/DBService.java

```diff
@@ -2624,36 +3017,46 @@ void updateSystemMetaFields(Domain domain, final String attribute, boolean delet
         }
     }
 
-    void updateRoleSystemMetaFields(Role role, final String attribute, boolean deleteAllowed, RoleSystemMeta meta) {
+    void updateRoleSystemMetaFields(Role role, final String attribute, RoleSystemMeta meta) {
 
         final String caller = "putrolesystemmeta";
 
         // system attributes we'll only set if they're available
         // in the given object
 
-        switch (attribute) {
-            case ZMSConsts.SYSTEM_META_AUDIT_ENABLED:
-                role.setAuditEnabled(meta.getAuditEnabled());
-                break;
-            default:
-                throw ZMSUtils.requestError("unknown role system meta attribute: " + attribute, caller);
+        if (ZMSConsts.SYSTEM_META_AUDIT_ENABLED.equals(attribute)) {
+            role.setAuditEnabled(meta.getAuditEnabled());
+        } else {
+            throw ZMSUtils.requestError("unknown role system meta attribute: " + attribute, caller);
+        }
+    }
+
+    void updateGroupSystemMetaFields(Group group, final String attribute, GroupSystemMeta meta) {
+
+        final String caller = "putgroupsystemmeta";
```

Comment:
I think that it will be preferable to place the content of this method inline in executePutGroupSystemMeta or make the caller an input argument.

This method is called only from executePutGroupSystemMeta so the "caller" is correct but if someone will reuse this method then he can accidentally leave this value as-is.